aws_crt_common_ws/aws-crt-c-flags/src/lib.rs

```diff
@@ -0,0 +1,469 @@
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// let build_info = CRTModuleBuildInfo::new("aws_crt_common_sys");
+    /// ```
+    pub fn new(module_name: &str) -> CRTModuleBuildInfo {
+        CRTModuleBuildInfo {
+            crt_module_name: module_name.parse().unwrap(),
+            crt_module_deps: vec![],
+            private_cflags: vec![],
+            public_cflags: vec![],
+            private_defines: vec![],
+            public_defines: vec![],
+            link_targets: vec![],
+            shared_lib: false,
+            lib_name: module_name.parse().unwrap(),
+            linker_path: Option::from(PathBuf::from(env::var_os("OUT_DIR").unwrap())),
+            include_dirs: vec![],
+            build_toolchain: Build::new(),
+        }
+    }
+
+    /// Declares other aws crt modules to have your sys package depend on. This is used for transitively
+    /// passing linker arguments and c-flags between different crates' builds of their sys modules.
+    ///
+    /// # Arguments
+    ///
+    /// * `dependency` - name of the crt sys crate you want to link your sys crate against. So for example,
+    ///                  if you're building aws-checksums, your sys crate would be aws_crt_checksums_sys, and you
+    ///                  would declare a dependency here on aws_crt_common_sys.
+    ///
+    /// # Examples
+    /// ```should_panic
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// let mut build_info = CRTModuleBuildInfo::new("aws_crt_checksums_sys");
+    /// build_info.add_module_dependency("aws_crt_common_sys");
+    /// ```
+    pub fn add_module_dependency(&mut self, dependency: &str) -> &mut CRTModuleBuildInfo {
+        let crt_module_env_var_res = env::var("CRT_MODULE_".to_owned() + dependency + "BUILD_CFG");
+
+        if crt_module_env_var_res.is_ok() {
+            let parse_res: Result<CRTModuleBuildInfo> =
+                serde_json::from_str(crt_module_env_var_res.unwrap().as_str());
+
+            if parse_res.is_ok() {
+                self.crt_module_deps.push(parse_res.unwrap());
+                return self;
+            }
+        }
+
+        panic!("Module: ".to_owned() + dependency + " does not appear to be a part of your dependency chain. Alternatively, the dependencies build script does not call CRTModuleBuildInfo::run_build() correctly");
+    }
+
+    /// Adds a c-flag to your build, and guarantees that when another module calls add_module_dependency()
+    /// on the module_name you used in ::new(), it will be transitively applied to their build.
+    ///
+    /// # Arguments
+    ///
+    /// * `c_flag` - compiler flag to apply to the build. Example `-fPIC`
+    ///
+    /// # Examples
+    /// ```should_panic
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// let mut build_info = CRTModuleBuildInfo::new("aws_crt_common_sys");
+    /// build_info.add_public_cflag("-fPIC");
+    /// ```
+    pub fn add_public_cflag(&mut self, c_flag: &str) -> &mut CRTModuleBuildInfo {
```

Comment:
nit: other Rust append builders frequently omit the add here (eg. `http::Request::builder().header("key", "value").build();`)

Reply:
Also, you can use `&mut Self` as the return type. It is a little bit less typing, but it depends on your personal preferences.